### PR TITLE
refactor: add `.clone(vault)` method to proxy-able BaseStrategy

### PR DESCRIPTION
**Pull request by fubuloubu**
This PR simplfiies the logic for creating a clone of a strategy that inherits the `BaseStrategyInitializable` proxy-enabled base class.

### CODE REVIEW

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -786,4 +788,22 @@ abstract contract BaseStrategyInitializable is BaseStrategy {
     ) external {
         _initialize(_vault, _strategist, _rewards, _keeper);
     }
+
+    function clone(address _vault) external returns (address newStrategy) {
```
Comment (poolpitako) [inline contracts/BaseStrategy.sol:792]: plz replace with:

```
    function clone(address _vault) external returns (address newStrategy) {
        newStrategy = this.clone(_vault, msg.sender, msg.sender, msg.sender);
    }

    function clone(
        address _vault,
        address _strategist,
        address _rewards,
        address _keeper
    ) external returns (address newStrategy) {
        // Copied from https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol
        bytes20 addressBytes = bytes20(address(this));

        assembly {
            // EIP-1167 bytecode
            let clone_code := mload(0x40)
            mstore(clone_code, 0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000)
            mstore(add(clone_code, 0x14), addressBytes)
            mstore(add(clone_code, 0x28), 0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000)
            newStrategy := create(0, clone_code, 0x37)
        }

        BaseStrategyInitializable(newStrategy).initialize(_vault, _strategist, _rewards, _keeper);

        emit Cloned(newStrategy);
    }
```

**Review by poolpitako — APPROVED** [review]

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -786,4 +788,31 @@ abstract contract BaseStrategyInitializable is BaseStrategy {
     ) external {
         _initialize(_vault, _strategist, _rewards, _keeper);
     }
+
+    function clone(address _vault) external returns (address) {
+        return clone(_vault, msg.sender, msg.sender, msg.sender);
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:793]:
```suggestion
        return this.clone(_vault, msg.sender, msg.sender, msg.sender);
```

**Comment by poolpitako** [comment]
Looks good 👍

**Comment by fubuloubu** [comment]
I couldn't figure out why the failure happens locally, rerunning CI to see if it's just being flakey

**Pull request merged**